src/Swarm/Game/Step.hs

```diff
@@ -1375,22 +1377,21 @@ execConst c vs s k = do
             let salvageInventory = E.union (target ^. robotInventory) (target ^. installedDevices)
             robotMap . at (target ^. robotID) . traverse . robotInventory .= salvageInventory
 
-            let salvageItems = concatMap (\(n, e) -> replicate n (e ^. entityName)) (E.elems salvageInventory)
-                numItems = length salvageItems
-
             -- Copy over the salvaged robot's log, if we have one
             inst <- use installedDevices
             em <- use entityMap
             creative <- use creativeMode
+            system <- use systemRobot
             logger <-
               lookupEntityName "logger" em
                 `isJustOr` Fatal "While executing 'salvage': there's no such thing as a logger!?"
             when (creative || inst `E.contains` logger) $ robotLog <>= target ^. robotLog
 
             -- Immediately copy over any items the robot knows about
-            -- but has 0 of
-            let knownItems = map snd . filter ((== 0) . fst) . elems $ salvageInventory
-            robotInventory %= \i -> foldr (insertCount 0) i knownItems
+            -- but has 0 of and in creative mode just get everything now.
+            let immediate = if system || creative then id else filter ((== 0) . fst)
+            let immediateItems = map snd . immediate . elems $ salvageInventory
+            robotInventory %= \i -> foldr (insertCount 0) i immediateItems
```

Comment:
It looks to me like when in system or creative mode, this will identify all items in the salvaged robot's inventory as "immediate" items, but then put them in the salvaging robot's inventory with a count of 0, regardless of how many there were in the salvaged robot's inventory.

Reply:
I don't even remember writing this code anymore. But it indeed looks wrong, I will rewrite it.